Sources/swift-build-prebuilts/main.swift

```diff
@@ -0,0 +1,437 @@
+                    for oldArtifact in library.artifacts {
+                        if !newArtifacts.contains(id: oldArtifact.id) {
+                            newArtifacts.insert(oldArtifact)
+                        }
+                    }
+                    newLibrary.artifacts = .init(newArtifacts.values)
+                    newLibraries.insert(newLibrary)
+                } else {
+                    newLibraries.insert(library)
+                }
+            }
+        }
+        let newManifest = Workspace.PrebuiltsManifest(libraries: .init(newLibraries.values))
+
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        let manifestData = try encoder.encode(newManifest)
+        let manifestFile = stageDir / "\(swiftVersion)-manifest.json"
+        try manifestFile.write(manifestData)
+    }
+}
+
+func canBuild(_ platform: Workspace.PrebuiltsManifest.Platform) -> Bool {
+    if dockeronly {
+        return platform.os == .linux
+    }
+    #if os(macOS)
+    if platform.os == .macos {
+        return true
+    }
+    #elseif os(Windows)
+    if platform.os == .windows {
+        return true
+    }
+    #endif
+    return docker && platform.os == .linux
+}
+
+func shell(_ command: String) throws {
+    let process = Process()
+#if os(Windows)
+    process.executableURL = URL(fileURLWithPath: "C:\\Windows\\System32\\cmd.exe")
+    process.arguments = ["/c", command]
+#else
+    process.executableURL = URL(fileURLWithPath: "/bin/bash")
+    process.arguments = ["-c", command]
+#endif
+    print("Running:", command)
+    try process.run()
+    process.waitUntilExit()
+    if process.terminationStatus != 0 {
+        throw StringError("failed: \(command)")
+    }
+}
+
+func downloadManifest(version: PrebuiltRepos.Version) async throws -> Workspace.PrebuiltsManifest? {
+    let manifestFile = swiftVersion + "-manifest.json"
+    let destination = await stageDir / manifestFile
+    if destination.exists {
+        do {
+            return try JSONDecoder().decode(
+                Workspace.PrebuiltsManifest.self,
+                from: try destination.read()
+            )
+        } catch {
+            // redownload it
+            try destination.remove()
+        }
+    }
+    let manifestURL = manifestHost.appending(components: version.tag, manifestFile)
+    print("Downloading:", manifestURL.absoluteString)
+    let httpClient = HTTPClient()
+    var headers = HTTPClientHeaders()
+    headers.add(name: "Accept", value: "application/json")
+    var request = HTTPClient.Request(kind: .generic(.get), url: manifestURL)
+    request.options.validResponseCodes = [200]
+
+    let response = try? await httpClient.execute(request) { _, _ in }
+    if let body = response?.body {
+        return try JSONDecoder().decode(
+            Workspace.PrebuiltsManifest.self,
+            from: body
+        )
+    }
+
+    return nil
+}
+
+// Path utility
```

Comment:
This is a script. Was playing with the idea here to see if it make scripts easier to write. Most of this thing is an experiment since it's not user facing. It won't be in the toolchain.

Reply:
I don't think the fact that it's a script should make any difference, we should still re-use the existing abstractions we have to avoid having to understand different sets of semantics from the rest of the codebase.

Reply:
OK, you convinced me :).